packages/blocks/src/api/raw-handling/paste-handler.js

```diff
@@ -244,10 +244,17 @@ export function pasteHandler( { HTML = '', plainText = '', mode = 'AUTO', tagNam
 		return htmlToBlocks( { html: piece, rawTransforms } );
 	} ) );
 
-	// If we're allowed to return inline content and there is only one block
+	// If `blocks` contains a single entry of one of these blocks,
+	// it can potentially be treated as an inline paste.
+	const inlineableBlocks = [
+		'core/paragraph',
```

Comment:
Could this be a block setting instead? We should also add it to any block that is just a text field, such as headings.

Reply:
Good suggestion, thanks @ellatrix! I've added that in 0b75d02c6.